src/SAML2/XML/saml/AuthnContext.php

```diff
@@ -0,0 +1,266 @@
+            );
+        }
+
+        $this->setAuthnContextClassRef($authnContextClassRef);
+
+        if (!is_null($authnContextDecl)) {
+            $this->setAuthnContextDecl($authnContextDecl);
+        }
+
+        if (!is_null($authnContextDeclRef)) {
+            $this->setAuthnContextDeclRef($authnContextDeclRef);
+        }
+
+        if (!empty($authenticatingAuthorities)) {
+            $this->setAuthenticatingAuthorities($authenticatingAuthorities);
+        }
+    }
+
+
+    /**
+     * Collect the value of the authnContextClassRef-property
+     *
+     * @return \SAML2\XML\saml\AuthnContextClassRef|null
+     */
+    public function getAuthnContextClassRef(): ?AuthnContextClassRef
+    {
+        return $this->authnContextClassRef;
+    }
+
+
+    /**
+     * Set the value of the authnContextClassRef-property
+     *
+     * @param \SAML2\XML\saml\AuthnContextClassRef|null $authnContextClassRef
+     * @return void
+     */
+    public function setAuthnContextClassRef(?AuthnContextClassRef $authnContextClassRef): void
+    {
+        $this->authnContextClassRef = $authnContextClassRef;
+    }
+
+
+    /**
+     * Collect the value of the authnContextDeclRef-property
+     *
+     * @return \SAML2\XML\saml\AuthnContextDeclRef|null
+     */
+    public function getAuthnContextDeclRef(): ?AuthnContextDeclRef
+    {
+        return $this->authnContextDeclRef;
+    }
+
+
+    /**
+     * Set the value of the authnContextDeclRef-property
+     *
+     * @param \SAML2\XML\saml\AuthnContextDeclRef|null $authnContextDeclRef
+     * @return void
+     */
+    public function setAuthnContextDeclRef(?AuthnContextDeclRef $authnContextDeclRef): void
+    {
+        $this->authnContextDeclRef = $authnContextDeclRef;
+    }
+
+
+    /**
+     * Collect the value of the authnContextDecl-property
+     *
+     * @return \SAML2\XML\saml\AuthnContextDecl|null
+     */
+    public function getAuthnContextDecl(): ?AuthnContextDecl
+    {
+        return $this->authnContextDecl;
+    }
+
+
+    /**
+     * Set the value of the authnContextDecl-property
+     *
+     * @param \SAML2\XML\saml\AuthnContextDecl|null $authnContextDecl
+     * @return void
+     */
+    public function setAuthnContextDecl(?AuthnContextDecl $authnContextDecl): void
+    {
+        $this->authnContextDecl = $authnContextDecl;
+    }
+
+
+    /**
+     * Collect the value of the authenticatingAuthorities-property
+     *
+     * @return \SAML2\XML\saml\AuthenticatingAuthority[]|null
+     */
+    public function getAuthticatingAuthorities(): ?array
+    {
+        return $this->authenticatingAuthorities;
+    }
+
+
+    /**
+     * Set the value of the authenticatingAuthorities-property
+     *
+     * @param \SAML2\XML\saml\AuthenticatingAuthority[]|null $authenticatingAuthorities
+     * @return void
+     */
+    public function setAuthenticatingAuthorities(?array $authenticatingAuthorities): void
+    {
+        $this->authenticatingAuthorities = $authenticatingAuthorities;
```

Comment:
Here I think we should check the elements and verify that all of them are `SAML2\XML\saml\AuthenticatingAuthority` objects.